daemon/api_system_recovery_keys.go

```diff
@@ -36,11 +39,50 @@ var systemRecoveryKeysCmd = &Command{
 	WriteAccess: rootAccess{},
 }
 
+// newRecoveryKeyAPISupported returns true if this system should use the new
+// FDE/recovery key APIs.
+//
+// TODO: usage of this function and the routes we support should be reviewed for
+// core26
+func newRecoveryKeyAPISupported(st *state.State) (bool, error) {
+	deviceCtx, err := snapstate.DeviceCtx(st, nil, nil)
+
+	// TODO: if we don't yet have a model, assume that it should be supported?
+	// is that the right thing to do? unsure if this is a realistic scenario,
+	// but it does happen in tests
+	if err != nil {
+		return true, nil
```

Comment:
no here we should return an error that is too early,  look for similar error about seeding, or at most return false but that is a bit confusing

Reply:
@pedronis  I think this needed for the install-time recovery key generation `InstallStepGenerateRecoveryKey`?

Reply:
it's probably fine to drop that check,  leaving a note there. But indeed that check would have to look at the used system model